src/algorithms/search/ternary_search.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+/**
+  *  Finds the maximum of unimodal function fn() within [left, right]
+  *  To find the minimum, revert the if/else statement or revert the comparison.
+  *  Time complexity: O(log(n))
+  */
+
+var ternarySearch = function (fn, left, right, precision) {
+  while (true) {
+    if (Math.abs(right - left) < precision)
+      return (left + right) / 2;
```

Comment:
I believe that moving this condition to the while loop like:
```javascript
while (Math.abs(right - left) > precision) {
 ...
}
return (left + right) / 2; 
```
Is more elegant than this ```while (true)```